### PR TITLE
Bluetooth: Coverity fixes

### PR DESCRIPTION
**Pull request by alwa-nordic**
This PR contains various fixes for issues discovered by Coverity.

### CODE REVIEW

File: subsys/bluetooth/host/l2cap.c [hunk 1]
```diff
@@ -2031,7 +2031,11 @@ static int l2cap_chan_le_send_sdu(struct bt_l2cap_le_chan *ch,
...
 	int ret;
 	size_t sent = 0;
-	size_t rem_len = buf->len;
+	size_t rem_len;
+
+	__ASSERT_NO_MSG(buf);
+
+	rem_len = buf->len;
 
 	while (buf && sent != rem_len) {
```
Comment (Thalley) [inline subsys/bluetooth/host/l2cap.c:2040]: With the assert I guess the `while(buf` check can be removed, as `buf` is never modified by the loop.

File: subsys/bluetooth/host/scan.c [hunk 2]
```diff
@@ -668,6 +668,19 @@ void bt_hci_le_adv_ext_report(struct net_buf *buf)
 			reassembling_advertiser.state = FRAG_ADV_DISCARDING;
 		}
 
+		if (evt->length > buf->len) {
+			LOG_WRN("Adv report corrupted (wants %d out of %d)", evt->length, buf->len);
```
Comment (Thalley) [inline subsys/bluetooth/host/scan.c:672]: Minor

```suggestion
			LOG_WRN("Adv report corrupted (wants %u out of %u)", evt->length, buf->len);
```

File: subsys/bluetooth/host/scan.c [hunk 3]
```diff
@@ -668,6 +668,19 @@ void bt_hci_le_adv_ext_report(struct net_buf *buf)
...
+
+			/* Start discarding irrespective of the `more_to_come` flag. We
+			 * assume we may have lost a partial adv report in the truncated
+			 * data.
+			 */
+			reassembling_advertiser.state = FRAG_ADV_DISCARDING;
+			net_buf_reset(buf);
+
+			return;
```
Comment (Thalley) [inline subsys/bluetooth/host/scan.c:681]: Should this also call `reset_reassembling_advertiser`? I'm unsure myself :D

File: subsys/bluetooth/host/l2cap.c (hunk 1 above)
Comment (alwa-nordic) [reply subsys/bluetooth/host/l2cap.c:2040]: Indeed. After fragment support was removed, this whole function appears useless. I'll remove it.

**Review by Thalley — DISMISSED** [review]

File: subsys/bluetooth/host/l2cap.c [hunk 4]
```diff
@@ -930,13 +929,20 @@ static void l2cap_chan_tx_process(struct k_work *work)
...
 		 * - a new SDU from the TX queue
 		 */
 		LOG_DBG("chan %p buf %p", ch, buf);
 
-		ret = l2cap_chan_le_send_sdu(ch, buf);
+		ret = l2cap_chan_le_send(ch, buf);
+
+		if (ret == rem_len) {
+			net_buf_unref(buf);
+		}
```
Comment (alwa-nordic) [inline subsys/bluetooth/host/l2cap.c:944]: There is some insane ownership logic going on here. `l2cap_chan_le_send` takes ownership of `buf` unless error or `ret == rem_len`. @jori-nordic Can you confirm I did this refactor right? I'm struggling to follow the `tx_buf` logic.

Comment (jori-nordic) [reply subsys/bluetooth/host/l2cap.c:944]: The loop does loop :/
I will refactor this when I have the time (lol)

**Review by jhedberg — DISMISSED** [review]

**Review by jori-nordic — COMMENTED** [review]
needs 🛠️

**Review by jori-nordic — CHANGES_REQUESTED** [review]
needs 🛠️

**Review by jori-nordic — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Pull request merged**